src/AutoRest.CSharp/Common/AutoRest/Plugins/CSharpProj.cs

```diff
@@ -91,96 +41,153 @@ internal static string GetVersion()
-            return true;
-        }
+                await autoRest.WriteFile(Path.Combine(Configuration.RelativeProjectFolder, filename), text, "source-file-csharp");
+            });
 
-        public void Execute(string defaultNamespace, string generatedDir, bool includeDfe, bool includeAzureKeyAuth, CSharpProjConfiguration config)
-        {
-            Execute(defaultNamespace, includeAzureKeyAuth, async (filename, text) =>
+        public void Execute()
+            => WriteCSProjFiles(async (filename, text) =>
             {
                 //TODO adding to workspace makes the formatting messed up since its a raw xml document
                 //somewhere it tries to parse it as a syntax tree and when it converts back to text
                 //its no longer valid xml.  We should consider a "raw files" concept in the work space
                 //so the file writing can still remain in one place
-                await File.WriteAllTextAsync(Path.Combine(config.AbsoluteProjectFolder, filename), text);
-            },
-                includeDfe, config);
-        }
+                await File.WriteAllTextAsync(Path.Combine(Configuration.AbsoluteProjectFolder, filename), text);
+            });
 
-        private void Execute(string defaultNamespace, bool includeAzureKeyAuth, Action<string, string> writeFile, bool includeDfe, CSharpProjConfiguration config)
+        private void WriteCSProjFiles(Action<string, string> writeFile)
         {
-            if (includeDfe)
+            // write src csproj
+            var csprojContent = Configuration.SkipCSProjPackageReference ? GetCSProj() : GetExternalCSProj();
+            writeFile($"{Configuration.Namespace}.csproj", csprojContent);
+
+            // write test csproj when needed
+            if (Configuration.MgmtTestConfiguration is not null)
             {
-                _coreCsProjContent += @"
-  <ItemGroup>
-    <PackageReference Include=""Azure.Core.Expressions.DataFactory"" />
-  </ItemGroup>";
+                var testCSProjContent = GetTestCSProj();
+                string testGenProjectFolder;
+                if (Configuration.MgmtTestConfiguration.OutputFolder is { } testGenProjectOutputFolder)
+                {
+                    testGenProjectFolder = Path.Combine(testGenProjectOutputFolder, "../");
+                }
+                else
+                {
+                    testGenProjectFolder = "../";
+                }
+                Console.WriteLine(Path.Combine(testGenProjectFolder, $"{Configuration.Namespace}.Tests.csproj"));
+                writeFile(FormatPath(Path.Combine(testGenProjectFolder, $"{Configuration.Namespace}.Tests.csproj")), testCSProjContent);
             }
+        }
+
+        private static string FormatPath(string? path)
+        {
+            if (string.IsNullOrEmpty(path))
+                return path ?? "";
+            return Path.GetFullPath(path.TrimEnd('/', '\\')).Replace("\\", "/");
+        }
```

Comment:
autorest uses JRPC to get interprocess messaging which works in the form of json, and the default path separator on windows `\` needs to be escaped otherwise on autorest side they got a JSON error because of the escaping.
Here we introduce this method to manually replace the symbol to `/` which does not requires an escaping.

We have the same method in mgmt test gen target, I just copied it here. We need a follow up there to clean things up: https://github.com/Azure/autorest.csharp/issues/4037